support-workers/src/main/scala/com/gu/zuora/ZuoraService.scala

```diff
@@ -29,9 +32,11 @@ class ZuoraService(val config: ZuoraConfig, client: FutureHttpClient, baseUrl: O
   def getAccount(accountNumber: String): Future[GetAccountResponse] =
     get[GetAccountResponse](s"accounts/$accountNumber", authHeaders)
 
-  def getAccountFields(identityId: IdentityId): Future[List[DomainAccount]] = {
+  def getAccountFields(identityId: IdentityId, now: OffsetDateTime): Future[List[DomainAccount]] = {
+    val recentDays = 28
+    val recently = now.minusDays(recentDays).format(DateTimeFormatter.ISO_ZONED_DATE_TIME)
     // WARNING constructing queries from strings is inherently dangerous.  Be very careful.
-    val queryData = QueryData(s"select AccountNumber, CreatedRequestId__c from account where IdentityId__c = '${identityId.value}'")
+    val queryData = QueryData(s"select AccountNumber, CreatedRequestId__c from account where IdentityId__c = '${identityId.value}' and UpdatedDate > '$recently'")
```

Comment:
I assume we're pretty confident this query will be more efficient than the previous version? Having not worked much with Zuora much is it reasonably safe to make adjustments to queries without being paranoid performance will be affected significantly?

Reply:
yes, this is a built in field, we would not worry too much about performance from that point of view.  I think it's indexed under the covers.
For custom fields we can choose whether they are indexed or not, and non indexed we wouldn't be able to query.  I believe we created CreatedRequestId__c as a non indexed field, hence we just iterate through.